### PR TITLE
update for tf2.4

### PR DESCRIPTION
**Pull request by fsx950223**
cc @mingxingtan 
I met some issues when I train the model with keras multi gpus on tf2.4, could you check it?
I wonder it's a bug of my environment or tensorflow.

### CODE REVIEW

File: efficientdet/keras/train.py [hunk 1]
```diff
@@ -74,7 +74,7 @@
 flags.DEFINE_integer('batch_size', 64, 'training batch size')
 flags.DEFINE_integer('eval_samples', 5000, 'The number of samples for '
                      'evaluation.')
-flags.DEFINE_integer('steps_per_execution', 1000,
+flags.DEFINE_integer('steps_per_execution', 1,
```
Comment (fsx950223) [inline efficientdet/keras/train.py:77]: Disable it in default since there is some issues in multi gpus training with uninitlized optimizer.

File: efficientdet/dataloader.py [hunk 2]
```diff
@@ -294,6 +305,21 @@ def dataset_parser(self, value, example_decoder, anchor_labeler, params):
...
+          input_processor.random_horizontal_flip()
+
+        input_processor.set_training_random_scale_factors(
+            params['jitter_min'], params['jitter_max'],
+            params.get('target_size', None))
+      else:
+        input_processor.set_scale_factors_to_output_size()
+
+      image = input_processor.resize_and_crop_image()
+      boxes, classes = input_processor.resize_and_crop_boxes()
```
Comment (fsx950223) [inline efficientdet/dataloader.py:321]: Resize image first could double speed up pipeline.

Comment (mingxingtan) [reply efficientdet/dataloader.py]: Interesting finding!

File: efficientdet/utils.py [hunk 3]
```diff
@@ -254,9 +254,7 @@ def batch_norm_class(is_training, strategy=None):
   if is_training and strategy == 'tpu':
     return TpuBatchNormalization
   elif is_training and strategy == 'gpus':
-    # TODO(fsx950223): use SyncBatchNorm after TF bug is fixed (incorrect nccl
-    # all_reduce). See https://github.com/tensorflow/tensorflow/issues/41980
-    return BatchNormalization
+    return SyncBatchNormalization
```
Comment (mingxingtan) [inline efficientdet/utils.py:257]: How about the speed of SyncBatchNormalization for multiple GPUs?

**Review by mingxingtan — APPROVED** [review]

File: efficientdet/utils.py [hunk 4]
```diff
@@ -622,13 +622,11 @@ def build_model_with_precision(pp, mm, ii, *args, **kwargs):
     inputs = tf.cast(ii, tf.bfloat16)
     with tf.tpu.bfloat16_scope():
       outputs = mm(inputs, *args, **kwargs)
-    set_precision_policy('float32')
```
Comment (fsx950223) [inline efficientdet/utils.py:632]: After remove 2 lines, I could train model with recompute_grad and mixed_precision.
Why set policy back to float32? Could I remove them? @mingxingtan

File: efficientdet/utils.py (hunk 3 above)
Comment (fsx950223) [reply efficientdet/utils.py:258]: It's slower about 40% than BatchNormalization for multiple GPUs. I believe it's acceptable.

Comment (fsx950223) [reply efficientdet/utils.py:258]: According to https://github.com/tensorflow/tensorflow/blob/9489702e35b16a40a1accf3b8b5ed557efae10c7/tensorflow/python/keras/layers/normalization_v2.py#L151.
Should I split replica_ctx.all_reduce?

File: efficientdet/utils.py (hunk 4 above)
Comment (mingxingtan) [reply efficientdet/utils.py:632]: Yes, please feel free to remove it. it is not necessary.

File: efficientdet/keras/train.py (hunk 1 above)
Comment (mingxingtan) [reply efficientdet/keras/train.py]: sgtm.

File: efficientdet/utils.py (hunk 3 above)
Comment (mingxingtan) [reply efficientdet/utils.py:258]: I did not quite understand these comments, but I think 40% slower is fine.

**Review by mingxingtan — APPROVED** [review]
